tooling/nargo_toml/src/semver.rs

```diff
@@ -0,0 +1,196 @@
+use crate::{errors::SemverError, ManifestError};
+use nargo::{
+    package::{Dependency, Package},
+    workspace::Workspace,
+};
+use semver::{Version, VersionReq};
+
+// Check that all of the packages in the workspace are compatible with the current compiler version
+pub(crate) fn semver_check_workspace(
+    workspace: Workspace,
+    current_compiler_version: String,
+) -> Result<(), ManifestError> {
+    let version = Version::parse(&current_compiler_version)
+        .expect("The compiler version is not a valid semver version");
```

Comment:
could this be a semver err?

Reply:
The compiler version is set by the noir compiler, so it would be a compiler error and not a user error